flags/decorators.py

```diff
@@ -4,18 +4,24 @@
 from flags.template_functions import flag_enabled
 
 
-def flag_required(flag_name, fallback_view=None, pass_if_set=True):
+def flag_check(flag_name, condition, fallback=None):
+    """ Check that a given flag has the given condition (True/False).
+    If the condition is not met, perform the fallback. """
     def decorator(func):
         def inner(request, *args, **kwargs):
             enabled = flag_enabled(request, flag_name)
 
-            if (enabled and pass_if_set) or (not enabled and not pass_if_set):
+            if (condition and enabled) or (not condition and not enabled):
                 return func(request, *args, **kwargs)
-            elif fallback_view is not None:
-                return fallback_view(request)
+            elif fallback is not None:
+                return fallback(request)
```

Comment:
Revisiting this code I wonder if this should be `fallback(request, *args, **kwargs)` in the case where a fallback view has some arguments. Perhaps out of scope for this PR.